external/mmdetection/detection_tasks/apis/detection/inference_task.py

```diff
@@ -218,15 +218,18 @@ def _add_predictions_to_dataset(self, prediction_results, dataset, confidence_th
                             polygon = Polygon(points=points)
                             if polygon.get_area() > 1e-12:
                                 shapes.append(Annotation(polygon, labels=labels, id=ID(f"{label_idx:08}")))
+                    if add_saliency_map:
+                        features = [features,
+                                    draw_instance_segm_saliency_map(shapes, dataset_item, self._labels)]
             else:
                 raise RuntimeError(
                     f"Detection results assignment not implemented for task: {self._task_type}")
 
             dataset_item.append_annotations(shapes)
-
-            if feature_vector is not None:
-                active_score = TensorEntity(name="representation_vector", numpy=feature_vector)
-                dataset_item.append_metadata_item(active_score, model=self._task_environment.model)
+            if features:
```

Comment:
When can `if features` evaluate to `False`?

Reply:
It can't be False, only None, if `dump_features=False`, expanded its condition